### PR TITLE
Filter Search Results By File Type

### LINKED ISSUE #31: Search filters by content type
Standard search UI should have some filters for types of content, including:
- Docs
- Spreadsheets
- Presentations
- Images

Location: maybe collapsed/expandable underneath the search box?

### PR DESCRIPTION
**Pull request by zenkimoto**
Closes #31 

### Description of Change
Add the ability to filter search by type of file.  On the main search page, when the user clicks on filter search, a modal will appear for the user to select the file types to filter by.  

An inventory was done on the current file types in the Content Library.  These file types are mapped to their respective mime types and added to the search query.

### Motivation and Context

Make it easier for someone searching for content to filter by file type.

### Checklist

- [ ] Ran `npm run lint` and updated code style accordingly
- [ ] `npm run test` passes
- [ ] PR has a description and all contributors/stakeholder are noted/cc'ed
- [ ] tests are updated and/or added to cover new code
- [ ] relevant documentation is changed and/or added

### CODE REVIEW

File: scripts/controllers/searchfilter_controller.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+
```
Comment (gunsch) [inline scripts/controllers/searchfilter_controller.ts:1]: naming nit:

`SearchFilterController` should be in a file named `search_filter_controller.ts`

**Review by gunsch — APPROVED** [review]
small changes!

**Comment by gunsch** [comment]
sorry this took so long, I think I started reviewing it a couple times and kept getting sidetracked

File: custom/styles/_custom.scss [hunk 2]
```diff
@@ -72,7 +71,19 @@
   background-color: #fff;
   box-shadow: 5px 5px 0px 0px $gray-25;
 
-  transition: .5s;
+  -webkit-transition: all 0.2s cubic-bezier(0.175, 0.885, 0.32, 1.275);
```
[the remark this answers is not in the document]
Comment (zenkimoto) [reply custom/styles/_custom.scss]: removed

**Pull request merged**